### PR TITLE
improve the connection selection

### PR DESCRIPTION
**Pull request by rphillips**

### CODE REVIEW

File: lib/driver.js [hunk 1]
```diff
@@ -988,8 +990,8 @@ PooledConnection.prototype.execute = function(query, args, executeCallback) {
 PooledConnection.prototype._getNextCon = function(unwrappedCallback) {
   var self = this,
       calledBack = false,
+      randomIndexes = _.shuffle(_.range(self.connections.length));
       timeoutTime = self.timeout * self.connections.length, // preserves legacy behavior regarding timeouts.
```
Comment (GeorgeJahad) [inline lib/driver.js]: if Math.random() is the problem, won't "shuffle()" also have a problem?  don't you think it is also using Math.random()?

**Comment by GeorgeJahad** [comment]
can we turn on this log message too, so we can confirm if this is a problem?

https://github.com/racker/ele/blob/master/node_modules/cassandra-client/lib/driver.js#L323

File: lib/driver.js (hunk 1 above)
Comment (rphillips) [inline lib/driver.js]: _.shuffle uses the Fisher Yates algorithm. Certainly better than the mod division in this code block which will lean towards 0 or the max length.

**Comment by rphillips** [comment]
That code block gets execute already, right?

**Comment by rphillips** [comment]
```javascript
var connectionLength = 8;
var _ = require('lodash');

for(var j = 0; j < 1000; j++) {
  var randomIndexes = _.shuffle(_.range(connectionLength));
  for (var i = 0; i < randomIndexes.length; i++) {
      process.stdout.write(randomIndexes[i].toString());
      process.stdout.write(',');
  }
  process.stdout.write('\n');
}
```

Looks pretty good to me.

**Comment by GeorgeJahad** [comment]
It looks to me like lodash shuffle() uses math.Random() internally too:
https://github.com/lodash/lodash/blob/a0a649aaeabb0b99300ecb32f80f060f3f39815d/vendor/underscore/underscore.js#L358
https://github.com/lodash/lodash/blob/a0a649aaeabb0b99300ecb32f80f060f3f39815d/vendor/underscore/underscore.js#L368
https://github.com/lodash/lodash/blob/a0a649aaeabb0b99300ecb32f80f060f3f39815d/vendor/underscore/underscore.js#L1399

File: lib/driver.js (hunk 1 above)
Comment (GeorgeJahad) [inline lib/driver.js]: Ah, good to know, thanks.

**Comment by GeorgeJahad** [comment]
The log code block gets executed, but I don't think the AEP's listen for the event to actually print to the logs.  In any case, I guess that is actually a PR against a different repo. so I'll +1.

**Comment by GeorgeJahad** [comment]
+1

**Pull request merged**